test/fixtures/class_all.js

```diff
@@ -194,6 +194,40 @@ class MyThing extends OtherThing {
      * @return {Promise<{newChannels: Channel[], foo: Bar}>} The Promise
      */
     promiseBar() {
+
+    /**
+     * Gets a Promise that will resolve with a generic function
+     *
+     * @return {Promise<Function>} The Promise
+     */
+    promiseGenericFunc() {
+    }
+
+    /**
+     * Gets a Promise that will resolve with a function with no arguments
+     * that returns a string.
+     *
+     * @return {Promise<Function(): string>} The Promise
+     */
+    promiseStringFunc() {
+    }
+
+    /**
+     * Gets a Promise that will resolve with a function with lots of arguments
+     * that returns an object.
+     *
+     * @return {Promise<Function(Array.<OtherThing>, object, number, string): object>} The Promise
+     */
+    promiseLotsArgsFunc() {
+    }
+
+    /**
+     * Gets a Promise that will resolve with a function with lots of arguments
+     * that returns the default type.
+     *
+     * @return {Promise<Function(Array.<OtherThing>, object, number, string)>} The Promise
+     */
+    promiseDefaultRetFunc() {
```

Comment:
Can you also add tests for:

```
@param {function(number): object}
@property {function(number): void}

@param {(value: string) => number}
@property {(value: string) => SomeComplexType}
@returns {(value: string) => void}
```

To test the different places it can go and the different supported syntax types? I would expect the second to also include the correct name.

Reply:
Added tests, but as mentioned not able to use the string parsing as we only receive "FUNCTION".